src/test/java/it/pagopa/swclient/mil/TerminalResourceTest.java

```diff
@@ -0,0 +1,126 @@
+
+    static TerminalEntity terminalEntity;
+
+    @BeforeAll
+    static void createTestObjects() {
+        commonHeader = TerminalTestData.getCorrectCommonHeader();
+        terminalDto = TerminalTestData.getCorrectTerminalDto();
+        terminalEntity = TerminalTestData.getCorrectTerminalEntity();
+    }
+
+    @Test
+    void testCreateTerminalEndpoint() {
+        Mockito.when(terminalRepository.persist(Mockito.any(TerminalEntity.class))).thenReturn(Uni.createFrom().item(terminalEntity));
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .header("RequestId", commonHeader.requestId())
+                .header("Authorization", commonHeader.authorization())
+                .and()
+                .body(terminalDto)
+                .when()
+                .post("/terminals")
+                .then()
+                .extract().response();
+
+        Assertions.assertEquals(201, response.statusCode());
+    }
+
+    @Test
+    void testCreateTerminalError_409() {
+        WriteError writeError = new WriteError(11000, "Duplicate key violation", new BsonDocument());
+        ServerAddress serverAddress = new ServerAddress("localhost", 27017);
+
+        Mockito.when(terminalRepository.persist(Mockito.any(TerminalEntity.class)))
+                .thenReturn(Uni.createFrom().failure(new MongoWriteException(writeError, serverAddress)));
+
+        // Effettua la richiesta API
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .header("RequestId", commonHeader.requestId())
+                .header("Authorization", commonHeader.authorization())
+                .and()
+                .body(terminalDto)
+                .when()
+                .post("/terminals")
+                .then()
+                .extract().response();
+
+        // Verifica che lo stato di risposta sia 409
+        Assertions.assertEquals(409, response.statusCode());
+    }
+
+    @Test
+    void testCreateTerminalError_500() {
+        Mockito.when(terminalRepository.persist(Mockito.any(TerminalEntity.class))).thenReturn(Uni.createFrom().failure(new WebApplicationException()));
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .header("RequestId", commonHeader.requestId())
+                .header("Authorization", commonHeader.authorization())
+                .and()
+                .body(terminalDto)
+                .when()
+                .post("/terminals")
+                .then()
+                .extract().response();
+
+        Assertions.assertEquals(500, response.statusCode());
+    }
+
+    @Test
+    void testCreateTerminalError_OtherCode() {
+        WriteError writeError = new WriteError(12345, "Some other error", new BsonDocument());
+        ServerAddress serverAddress = new ServerAddress("localhost", 27017);
+
+        Mockito.when(terminalRepository.persist(Mockito.any(TerminalEntity.class)))
```

Comment:
same as above, mock service

Reply:
Resolved in [https://github.com/pagopa/mil-terminal-registry/pull/25/commits/8c55df2932810878f744b7992b7a535c9c819587](https://github.com/pagopa/mil-terminal-registry/pull/url)